x-pack/plugins/alerting/server/alerts_service/lib/set_alerts_to_untracked.ts

```diff
@@ -155,21 +155,26 @@ const ensureAuthorizedToUntrack = async (params: SetAlertsToUntrackedParamsWithD
 };
 
 const getAuthorizedAlertsIndices = async ({
-  featureIds,
-  getAuthorizedRuleTypes,
+  ruleTypeIds,
+  getAllAuthorizedRuleTypesFindOperation,
   getAlertIndicesAlias,
   spaceId,
   logger,
 }: SetAlertsToUntrackedParamsWithDep) => {
   try {
-    const authorizedRuleTypes =
-      (await getAuthorizedRuleTypes?.(AlertingAuthorizationEntity.Alert, new Set(featureIds))) ||
-      [];
-    const indices = getAlertIndicesAlias?.(
-      authorizedRuleTypes.map((art: { id: string }) => art.id),
-      spaceId
-    );
-    return indices;
+    const authorizedRuleTypes = await getAllAuthorizedRuleTypesFindOperation?.({
+      authorizationEntity: AlertingAuthorizationEntity.Alert,
+      ruleTypeIds,
+    });
```

Comment:
I think it depends on the use case. `ruleTypeIds` is an optional field and if omitted it will return all rule types the user has access to. If we add `[]` (as before) we will not get any authorized rule types at all. I changed the behavior because I thought that if you do not provide any `ruleTypeIds` you should get all the ones you have access too. Note that this is only used for filtering and it does not affect the authorization in any way.